src/pydash/arrays.py

```diff
@@ -90,8 +96,13 @@
     "zip_with",
 )
 
+T = t.TypeVar("T")
+T2 = t.TypeVar("T2")
+SequenceT = t.TypeVar("SequenceT", bound=t.Sequence)
+MutableSequenceT = t.TypeVar("MutableSequenceT", bound=t.MutableSequence)
+
 
-def chunk(array, size=1):
+def chunk(array: t.List[T], size: int = 1) -> t.List[t.List[T]]:
```

Comment:
It's been some time since I've done much with Python typing, but I'm not sure about requiring the arguments to be an iterable of the same type (e.g. `foo(t.List[T]) -> t.List[t.List[T]]`). Unless specified, there usually aren't restrictions/requirements that the iterable contain a single type. Or am I misunderstanding how `[T]` works where it's ok if, e.g., a list has a mixture of strings and integers? Or would `chunk([1, 'a'])` fail this type hint and we need something like less restrictive like `foo(t.Iterable) -> t.Iterable[t.Iterable]`?

Reply:
Yes a type variable (`T` here) can be a union of types.
So if we take your example when passing `[1, 'a']` to the `chunk` function `T` would be equal to `Union[int, str]` so the full computed type of the function in this case would be `(t.List[Union[int, str]], int) -> t.List[t.List[Union[int, str]]]`

```py
# type_test.py
import typing as t
import pydash as _


my_list: t.List[t.Union[int, str]] = [1, 'a']

x = _.chunk(my_list)

reveal_type(x)
```
```bash
$ mypy type_test.py
type_test.py:9:13: note: Revealed type is "builtins.list[builtins.list[Union[builtins.int, builtins.str]]]"
Success: no issues found in 1 source file
```